packages/block-editor/src/components/link-control/link-preview.js

```diff
@@ -7,49 +7,99 @@ import classnames from 'classnames';
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { Button, ExternalLink } from '@wordpress/components';
+import {
+	Button,
+	ExternalLink,
+	__experimentalText as Text,
+} from '@wordpress/components';
 import { filterURLForDisplay, safeDecodeURI } from '@wordpress/url';
+import { Icon, globe } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
 import { ViewerSlot } from './viewer-slot';
 
+import useRemoteUrlData from './use-remote-url-data';
+
 export default function LinkPreview( { value, onEditClick } ) {
+	const { richData, isFetching } = useRemoteUrlData( value?.url );
+
+	// Rich data may be an empty object so test for that.
+	const hasRichData = richData && Object.keys( richData ).length;
```

Comment:
Should we just move this check inside the hook instead?

Reply:
Yes